pkg/controller/pod_volume_restore_controller.go

```diff
@@ -111,98 +111,84 @@ func NewPodVolumeRestoreController(
 
 func (c *podVolumeRestoreController) pvrHandler(obj interface{}) {
 	pvr := obj.(*arkv1api.PodVolumeRestore)
-	log := c.logger.WithField("key", kube.NamespaceAndName(pvr))
+	log := loggerForPodVolumeRestore(c.logger, pvr)
 
-	if !shouldEnqueuePVR(pvr, c.podLister, c.nodeName, log) {
+	if !isPVRNew(pvr) {
+		log.Debugf("Restore is not new, not enqueuing")
 		return
 	}
 
-	log.Debug("enqueueing")
-	c.enqueue(obj)
-}
-
-func (c *podVolumeRestoreController) podHandler(obj interface{}) {
-	pod := obj.(*corev1api.Pod)
-	log := c.logger.WithField("key", kube.NamespaceAndName(pod))
-
-	for _, pvr := range pvrsToEnqueueForPod(pod, c.podVolumeRestoreLister, c.nodeName, log) {
-		c.enqueue(pvr)
+	pod, err := c.podLister.Pods(pvr.Spec.Pod.Namespace).Get(pvr.Spec.Pod.Name)
+	if err != nil {
```

Comment:
I think we should log not-found errors at debug level, and all other errors at error level. WDYT?